pkg/controllers/drift/controller.go

```diff
@@ -0,0 +1,137 @@
+package drift
```

Comment:
Have we load tested this at all? I have some concern how frequently we may be hitting the APIs considering how many different changes can enqueue requests on this controller.

Reply:
I haven't personally, but I think most of the calls in the `IsNodeDrifted` are being called from a cache or the `kubeClient.Get`. Is there anything that you see which can be a bottleneck in terms of api call ?
I would require assistance from someone in Karpenter team to load test this one.

Reply:
If there are perf concerns, let's set `MaxConcurrentReconciles: 1`